frontend/src/static/js/services/webstatus-firebase-auth-service.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {customElement, property, state} from 'lit/decorators.js';
+import {consume, provide} from '@lit/context';
+import {
+  FirebaseApp,
+  firebaseAppContext,
+} from '../contexts/firebase-app-context.js';
+import {
+  AuthConfig,
+  firebaseAuthContext,
+} from '../contexts/firebase-auth-context.js';
+import {
+  Auth,
+  GithubAuthProvider,
+  connectAuthEmulator,
+  getAuth,
+  signInWithPopup,
+} from 'firebase/auth';
+import {User, firebaseUserContext} from '../contexts/firebase-user-context.js';
+import {ServiceElement} from './service-element.js';
+
+interface FirebaseAuthSettings {
+  emulatorURL: string;
+}
+
+@customElement('webstatus-firebase-auth-service')
+export class WebstatusFirebaseAuthService extends ServiceElement {
+  @property({type: Object})
+  settings!: FirebaseAuthSettings;
+
+  @consume({context: firebaseAppContext, subscribe: true})
+  @state()
+  firebaseApp?: FirebaseApp;
+
+  @provide({context: firebaseAuthContext})
+  firebaseAuthConfig?: AuthConfig;
+
+  @provide({context: firebaseUserContext})
+  user?: User;
+
+  initialized: boolean = false;
+
+  authInitializer: (app: FirebaseApp | undefined) => Auth = getAuth;
```

Comment:
If this always returns `getAuth`, then why not just pass in `getAuth` on lines 67 and 68?

Reply:
I did that so that I can change it for testing with the stub.